src/jit/lower.cpp

```diff
@@ -381,8 +383,44 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
         }
         break;
     case GT_CALL:
-        NYI("Call with TYP_LONG return value");
-        break;
+        {
+            GenTree* parent = data->parent;
+            if (parent != nullptr)
```

Comment:
I see that we don't need to force IR to be var=call if GT_CALL is a top-level node and hence the check parent != nullptr.  It would be useful to add a comment line explaining that here.